src/Model/Cms/AddCompetitionDTO.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace BalticRobo\Api\Model\Cms;
+
+
+use Symfony\Component\HttpFoundation\Request;
+
+class AddCompetitionDTO
+{
+    private $display_name;
+    private $name;
+    private $year;
+    private $registartionType;
+
+    private function __construct()
+    {
+    }
+
+    public static function createFromRequestAndYear(Request $request, int $year): self{
+        $content = json_decode($request->getContent(), true);
+        $dto = new self();
+
+        $dto->setDisplayName($content['display_name']);

```

Comment:
Avoid setters here too. You are inside object so you can set private properties as well.

```suggestion
        $dto->displayName = $content['display_name'];
```